core/src/main/scala/org/http4s/Scheme.scala

```diff
@@ -0,0 +1,59 @@
+package org.http4s
+
+import cats.{Order, Show}
+ import org.http4s.internal.parboiled2.CharPredicate.{Alpha, Digit}
+import org.http4s.internal.parboiled2.{Parser => PbParser}
+import org.http4s.parser.Http4sParser
+import org.http4s.util.{hashLower, Writer}
+
+final class Scheme private (val value: String) extends Comparable[Scheme] {
+  override def equals(o: Any) = o match {
+    case that: Scheme => this.value.equalsIgnoreCase(that.value)
+    case _ => false
+  }
+
+  private[this] var hash = 0
+  override def hashCode(): Int = {
```

Comment:
This is going to get repetitive.  We could have a CaseInsensitiveStringLike, but I'm trying to move away from inheritance.  I don't think we can abstract the equals without lugging around a manifest, because erasure.